File: backoffice/src/shared/domain/StringValueObject.py
"""
 *
 * Classes 
 *
"""

class StringValueObject:

    """
     *
     * Parameters 
     *
    """

    _value : str

    """
     *
     * Methods 
     *
    """

    def __init__( self, value : str ) -> None:
        self._value = value
    
    def value( self ) -> str:
        return self._value

    def isEmpty( self ) -> bool:
        if self._value == '':
            return True
        return False
    
    def equals( self, value : str ) -> bool:
        if self._value == value:
            return True
        return False

File: attention/src/shared/domain/valueObject/StringValueObject.cs
/* 
 * 
 * Classes 
 *
*/

namespace attention.src.shared.domain {

    public class StringValueObject {

        /* 
         * 
         * Attributes
         *
        */

        private string _value;

        /* 
         * 
         * Methods
         *
        */

        public StringValueObject( string value ) {
            _value = value;
        }

        public string value() {
            return _value;
        }

        public bool isEmpty() {
            return _value == string.Empty;
        }

        public bool equals( StringValueObject value ) {
            return _value == value.value();
        }

    }

}